### PR TITLE
middleware ordering added to doc

### PR DESCRIPTION
**Pull request by rachelappel**
#5172

### CODE REVIEW

**Comment by rachelappel** [comment]
@Tratcher Take a look? I added the ordering to the table. I completely forget when some run so I left those blank.

File: aspnetcore/fundamentals/middleware.md [hunk 1]
```diff
@@ -188,17 +188,17 @@ app.Map("/level1/level2", HandleMultiSeg);
...
-| [Static Files](xref:fundamentals/static-files) | Provides support for serving static files and directory browsing. |
+ASP.NET Core ships with the following middleware components, as well as a description of the order in which they run:
+
+| Middleware | Description | Ordering |
+| ----- | ------- | ------- |
+| [Authentication](xref:security/authentication/identity) | Provides authentication support. | Before `HttpContext.User` is needed. Terminal for OAuth callbacks. |
+| [CORS](xref:security/cors) | Configures Cross-Origin Resource Sharing. | Before components that use CORS |
+| [Response Caching](xref:performance/caching/middleware) | Provides support for caching responses. | Before components that want caching |
+| [Response Compression](xref:performance/response-compression) | Provides support for compressing responses. | Before components that want to use compression |
+| [Routing](xref:fundamentals/routing) | Defines and constrains request routes. |  |
```
Comment (Tratcher) [inline aspnetcore/fundamentals/middleware.md]: Terminal for matching routes.

File: aspnetcore/fundamentals/middleware.md [hunk 2]
```diff
@@ -188,17 +188,17 @@ app.Map("/level1/level2", HandleMultiSeg);
...
+| [Session](xref:fundamentals/app-state) | Provides support for managing user sessions. |  |
```
Comment (Tratcher) [inline aspnetcore/fundamentals/middleware.md]: Before components that want to use Session.

File: aspnetcore/fundamentals/middleware.md [hunk 3]
```diff
@@ -188,17 +188,17 @@ app.Map("/level1/level2", HandleMultiSeg);
...
+| [Static Files](xref:fundamentals/static-files) | Provides support for serving static files and directory browsing. | Terminal if a request matches files |
 | [URL Rewriting Middleware](xref:fundamentals/url-rewriting) | Provides support for rewriting URLs and redirecting requests. |
```
Comment (Tratcher) [inline aspnetcore/fundamentals/middleware.md]: Before components that consume the URL.

Why does this one include `Middleware` when the others don't?

File: aspnetcore/fundamentals/middleware.md [hunk 4]
```diff
@@ -188,17 +188,17 @@ app.Map("/level1/level2", HandleMultiSeg);
...
-| [Response Compression](xref:performance/response-compression) | Provides support for compressing responses. |
-| [Routing](xref:fundamentals/routing) | Defines and constrains request routes. |
-| [Session](xref:fundamentals/app-state) | Provides support for managing user sessions. |
-| [Static Files](xref:fundamentals/static-files) | Provides support for serving static files and directory browsing. |
+ASP.NET Core ships with the following middleware components, as well as a description of the order in which they run:
+
+| Middleware | Description | Ordering |
+| ----- | ------- | ------- |
+| [Authentication](xref:security/authentication/identity) | Provides authentication support. | Before `HttpContext.User` is needed. Terminal for OAuth callbacks. |
+| [CORS](xref:security/cors) | Configures Cross-Origin Resource Sharing. | Before components that use CORS |
```
Comment (Tratcher) [inline aspnetcore/fundamentals/middleware.md]: Diagnostics is missing from the list. So is HttpOverrides/ForwardedHeaders, RequestLocalization, and WebSockets.

File: aspnetcore/fundamentals/middleware.md (hunk 3 above)
Comment (rachelappel) [reply aspnetcore/fundamentals/middleware.md]: "Why does this one include Middleware when the others don't?" 
My guess: The author was probably moonlighting at the Department of Redundancy Department at the time. I can remove it.

Comment (guardrex) [reply aspnetcore/fundamentals/middleware.md]: We had a reason for doing it :point_right: https://github.com/aspnet/Docs/pull/2494 (January 11, 2017)

A year later tho, I don't recall why. I think Tom and I spoke about it off the PR ... @tdykstra do you recall why we did that?

Also note that it's that way in the TOC, so u could change it back there, too.

File: aspnetcore/fundamentals/middleware.md [hunk 5]
```diff
@@ -188,18 +188,22 @@ app.Map("/level1/level2", HandleMultiSeg);
...
+| [Response Caching](xref:performance/caching/middleware) | Provides support for caching responses. | Before components that want caching |
+| [Response Compression](xref:performance/response-compression) | Provides support for compressing responses. | Before components that want to use compression |
+| [Routing](xref:fundamentals/routing) | Defines and constrains request routes. | Terminal for matching routes. |
+| [Session](xref:fundamentals/app-state) | Provides support for managing user sessions. | Before components that want to use Session. |
+| [Static Files](xref:fundamentals/static-files) | Provides support for serving static files and directory browsing. | Terminal if a request matches files |
+| [URL Rewriting ](xref:fundamentals/url-rewriting) | Provides support for rewriting URLs and redirecting requests. | Before components that consume the URL. |
+| [Diagnostics](xref:security/) | Configure diagnostics. |  |
+| [HttpOverrides/ForwardedHeaders](xref:security/) | x. |  |
+| [RequestLocalization](xref:security/) | Provides localization support. |  |
+| [WebSockets](xref:security/) | Enables the WebSockets protocol. |  |
```
Comment (Tratcher) [inline aspnetcore/fundamentals/middleware.md]: links are wrong

File: aspnetcore/fundamentals/middleware.md [hunk 6]
```diff
@@ -188,18 +188,22 @@ app.Map("/level1/level2", HandleMultiSeg);
...
+| [CORS](xref:security/cors) | Configures Cross-Origin Resource Sharing. | Before components that use CORS |
+| [Response Caching](xref:performance/caching/middleware) | Provides support for caching responses. | Before components that want caching |
+| [Response Compression](xref:performance/response-compression) | Provides support for compressing responses. | Before components that want to use compression |
+| [Routing](xref:fundamentals/routing) | Defines and constrains request routes. | Terminal for matching routes. |
+| [Session](xref:fundamentals/app-state) | Provides support for managing user sessions. | Before components that want to use Session. |
+| [Static Files](xref:fundamentals/static-files) | Provides support for serving static files and directory browsing. | Terminal if a request matches files |
+| [URL Rewriting ](xref:fundamentals/url-rewriting) | Provides support for rewriting URLs and redirecting requests. | Before components that consume the URL. |
+| [Diagnostics](xref:security/) | Configure diagnostics. |  |
```
Comment (Tratcher) [inline aspnetcore/fundamentals/middleware.md]: Before components that generate errors

File: aspnetcore/fundamentals/middleware.md [hunk 7]
```diff
@@ -188,18 +188,22 @@ app.Map("/level1/level2", HandleMultiSeg);
...
+| [HttpOverrides/ForwardedHeaders](xref:security/) | x. |  |
```
Comment (Tratcher) [inline aspnetcore/fundamentals/middleware.md]: Before components that consume the fields being updated (e.g. Scheme, Host, ClientIP, Method)

File: aspnetcore/fundamentals/middleware.md [hunk 8]
```diff
@@ -188,18 +188,22 @@ app.Map("/level1/level2", HandleMultiSeg);
...
+| [RequestLocalization](xref:security/) | Provides localization support. |  |
```
Comment (Tratcher) [inline aspnetcore/fundamentals/middleware.md]: Before localization sensitive components.

File: aspnetcore/fundamentals/middleware.md (hunk 5 above)
Comment (Tratcher) [reply aspnetcore/fundamentals/middleware.md]: Before components that want to accept WebSocket requests.

File: aspnetcore/fundamentals/middleware.md (hunk 6 above)
Comment (Tratcher) [reply aspnetcore/fundamentals/middleware.md]: The list was previously alphabetical.

File: aspnetcore/fundamentals/middleware.md [hunk 9]
```diff
@@ -188,18 +188,22 @@ app.Map("/level1/level2", HandleMultiSeg);
...
-| ----- | ------- |
-| [Authentication](xref:security/authentication/identity) | Provides authentication support. |
-| [CORS](xref:security/cors) | Configures Cross-Origin Resource Sharing. |
-| [Response Caching](xref:performance/caching/middleware) | Provides support for caching responses. |
-| [Response Compression](xref:performance/response-compression) | Provides support for compressing responses. |
-| [Routing](xref:fundamentals/routing) | Defines and constrains request routes. |
-| [Session](xref:fundamentals/app-state) | Provides support for managing user sessions. |
-| [Static Files](xref:fundamentals/static-files) | Provides support for serving static files and directory browsing. |
-| [URL Rewriting Middleware](xref:fundamentals/url-rewriting) | Provides support for rewriting URLs and redirecting requests. |
+ASP.NET Core ships with the following middleware components, as well as a description of the order in which they run:
```
Comment (Tratcher) [inline aspnetcore/fundamentals/middleware.md]: the order in which they should be added:

**Comment by guardrex** [comment]
[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware?branch=pr-en-us-5184&tabs=aspnetcore2x#built-in-middleware)

I linked `ForwardedHeaders/HttpOverrides` to API content. Is there a better option?

**Comment by rachelappel** [comment]
@guardrex  Looks good

**Pull request merged**